tuplex/core/src/physical/TransformTask.cc

```diff
@@ -543,43 +543,40 @@ namespace tuplex {
 
         auto functor = reinterpret_cast<codegen::read_block_exp_f>(_functor);
 
-        auto numInputExceptions = _inputExceptionInfo.numExceptions;
-        auto inputExceptionIndex = _inputExceptionInfo.exceptionIndex;
-        auto inputExceptionRowOffset = _inputExceptionInfo.exceptionRowOffset;
-        auto inputExceptionByteOffset = _inputExceptionInfo.exceptionByteOffset;
-
-        // First, prepare the input exception partitions to pass into the code-gen
-        // This is done to simplify the LLVM code. We will end up passing it an
-        // array of expPtrs which point to the first exception in their partition
-        // and expPtrSizes which tell how many exceptions are in that partition.
-        auto arrSize = _inputExceptions.size() - inputExceptionIndex;
-        auto expPtrs = new uint8_t*[arrSize];
-        auto expPtrSizes = new int64_t[arrSize];
-        int expInd = 0;
-        // Iterate through all exception partitions beginning at the one specified by the starting index
-        for (int i = inputExceptionIndex; i < _inputExceptions.size(); ++i) {
-            auto numRows = _inputExceptions[i]->getNumRows();
-            auto ptr = _inputExceptions[i]->lock();
-            // If its the first partition, we need to account for the offset
-            if (i == inputExceptionIndex) {
-                numRows -= inputExceptionRowOffset;
-                ptr += inputExceptionByteOffset;
-            }
-            expPtrSizes[expInd] = numRows;
-            expPtrs[expInd] = (uint8_t *) ptr;
-            expInd++;
-        }
+        int64_t totalNormalRowCounter = 0;
+        int64_t totalGeneralRowCounter = 0;
+        int64_t totalFallbackRowCounter = 0;
+        int64_t totalFilterCounter = 0;
+
+        uint8_t **generalPartitions = new uint8_t*[_generalPartitions.size()];
```

Comment:
why not use vector here? then we have RAII and do not have to worry about leaks...